src/ui/organisms/filePicker/FilePicker.tsx

```diff
@@ -0,0 +1,121 @@
+import { areFilesAccepted, asMutable, toReadableFileSize } from 'utils'
+import { useFileDispatch, useFileSelector } from 'hook/storeHook/fileHook'
+import { storeFiles, removeFile, removeAllFiles, getFiles } from 'domain/file'
+import type { ThunkResult } from 'domain/file/store/store'
+import type { FileDescriptor } from 'domain/file/store/selector/file.selector'
+import { useTranslation } from 'hook/useTranslation'
+import type { UseTranslationResponse } from 'hook/useTranslation'
+import { FileInput } from 'ui/atoms/fileInput/FileInput'
+import type { FileInputProps } from 'ui/atoms/fileInput/FileInput'
+import { Icon } from 'ui/atoms/icon/Icon'
+import { ListItem } from 'ui/atoms/listItem/ListItem'
+import { Typography } from 'ui/atoms/typography/Typography'
+import { List } from 'ui/atoms/list/List'
+import './filePicker.scss'
+import './i18n/index'
+
+export type FilePickerProps = Pick<
+  FileInputProps,
+  'label' | 'description' | 'multiple' | 'acceptedFormats' | 'size'
+> & {
+  /**
+   * Displays a clickable text to delete all selected files.
+   */
+  readonly clearAll?: boolean
+}
+
+// eslint-disable-next-line max-lines-per-function
+export const FilePicker: React.FC<FilePickerProps> = ({
+  clearAll = true,
+  ...props
+}: DeepReadonly<FilePickerProps>) => {
+  const { t }: UseTranslationResponse = useTranslation()
+  const fileDispatch = useFileDispatch()
+  const fileList: FileDescriptor[] = useFileSelector(getFiles)
+  const [isError, setError]: UseState<boolean> = useState(false)
+  const [errorMessage, setErrorMessage]: UseState<string> = useState('')
+
+  const displayFileSize = (size: number): string => {
+    const { value, unit }: ReadableFileSize = toReadableFileSize(size, (size: number) =>
+      Number(size.toFixed(2))
+    )
+    return `${value} ${t(`filePicker:filePicker.fileSizeUnit.${unit}`)}`
+  }
+
+  const handleDropped = useCallback(
+    (files: DeepReadonly<File[]>) => {
+      if (areFilesAccepted(files, props.acceptedFormats)) {
+        setError(false)
+        fileDispatch(
+          storeFiles(
+            files.map((file: DeepReadonly<File>) => {
+              return {
+                id: short.generate(),
+                name: file.name,
+                size: file.size,
+                type: file.type,
+                stream: file.stream() as unknown as ReadableStream
+              }
+            })
+          )
+        )
+      } else {
+        setError(true)
+        setErrorMessage(t(`filePicker:filePicker.errorMessage.type`))
+      }
+    },
+    [props.acceptedFormats, fileDispatch, t]
+  )
+
+  const handleRemove = useCallback(
+    (id: string) => (): ThunkResult<Promise<void>> => fileDispatch(removeFile(id)),
+    [fileDispatch]
+  )
+
+  const handleRemoveAll = useCallback(
+    (): ThunkResult<Promise<void>> => fileDispatch(removeAllFiles()),
+    [fileDispatch]
+  )
+
+  const RemoveIcon = (id: string): JSX.Element => (
```

Comment:
Concerning the naming, it seemed to me that we said that we write in PascalCase when the component is then called like this `<MyComponent id={id1} />` and that we use the camelCase when we call it like this: `myComponent(id1)`, or I am wrong?
I have the impression that here it is a mix of both.

Reply:
Yes maybe it is better to write : 
```tsx
const RemoveIcon = ({ id }: Readonly<{ id: string }>): JSX.Element => (
    <div className="okp4-file-picker-list-item-delete" onClick={handleRemove(id)}>
      <Icon invertColor name="close" />
    </div>
  )
  
  ...
  
  rightElement={<RemoveIcon id={id} />}
  ```
  
  what do you think @lolottetheclash ?

Reply:
Yeah, I think it's better too. 
But I wonder if we should make a type for this: `Readonly<{ id: string }>` to become this:
`Readonly<RemoveIconProps>`

`type RemoveIconProps = {
  id : string
}`

or if it's ok to leave it like that... @ccamel @fredericvilcot ?

Reply:
For an internal component, I would leave it like that.

Reply:
Good to know ! thanks ;)